cpp/benchmarks/tpch/utils.hpp

```diff
@@ -430,28 +534,105 @@ int32_t days_since_epoch(int year, int month, int day)
   return static_cast<int32_t>(diff);
 }
 
-struct tpch_example_args {
-  std::string dataset_dir;
-  std::string memory_resource_type;
+/**
+ * @brief Read the scale factor from the environment variable `CUDF_TPCH_SF`
+ */
+cudf::size_type get_sf()
+{
+  char* val          = getenv("CUDF_TPCH_SF");
+  cudf::size_type sf = (val == NULL) ? 1 : atoi(val);
```

Comment:
C++ should prefer comparing to `nullptr` instead of `NULL`.

Also this can be `const`.

Also use `std::atoi` rather than `atoi`. However, did we decide that scale factors are always integers? I thought we discussed the possibility of scale factors smaller than 1, in which case we would use a double or float. If that is correct, let's use `std::atof` and return `double` instead of `cudf::size_type`. Even if we don't support fractional scale factors, this should return `int` and not `cudf::size_type` because this is representing a generic integer (`int`) and not a cudf table size (`cudf::size_type`).
```suggestion
  cudf::size_type const sf = (val == nullptr) ? 1 : std::atoi(val);
```